scripts/zkai_paper_claim_pack_gate.py

```
@@ -0,0 +1,354 @@
+
+
+def _assert_evidence_paths_exist(payload: dict[str, Any]) -> None:
+    refs = payload.get("evidence_refs")
+    if not isinstance(refs, list) or not refs:
+        raise ClaimPackGateError("evidence_refs must be a non-empty list")
+    for index, ref in enumerate(refs):
+        if not isinstance(ref, dict):
+            raise ClaimPackGateError(f"evidence_refs[{index}] must be an object")
+        path = ref.get("path")
+        if not isinstance(path, str) or path.startswith("/") or ".." in pathlib.PurePosixPath(path).parts:
+            raise ClaimPackGateError(f"evidence_refs[{index}].path must be repo-relative")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)